### ISSUE TITLE
[Question] Chapter14 gan example needs more explanation

### ISSUE THREAD

**Issue by QiXuanWang** [zackchase/mxnet-the-straight-dope#439]
I don't understand below:
```Python
        with autograd.record():
            real_output = netD(data)
            errD_real = loss(real_output, real_label)

            fake = netG(noise)
            fake_output = netD(fake.detach())   
            errD_fake = loss(fake_output, fake_label)
            errD = errD_real + errD_fake
            errD.backward()
```
For netD(fake.detach()), my own example failed here due to incompatible data shape. It seems netG's dense layer is not arbitrary? And why using detach() ?

Also, please add more information if you plan to give introduction on GAN.
Here: 
> Generator * Draw some parameter zz from a source of randomness, e.g. a normal distribution z∼N(0,1)z∼N(0,1). * Apply a function ff such that we get x′=G(u,w)x′=G(u,w) * Compute the gradient with respect to ww to minimize logp(y=fake|x′)log⁡p(y=fake|x′)

I don't see where z is used and why suddenly u,w. I know it's simple if you look up GAN paper but, I suppose you want to teach GAN a little bit?
